site/viamguides/src/environment-variables/environment-variables.md

```diff
@@ -0,0 +1,223 @@
+ ![Viam app](assets/ViamRover.png)
+
+<!-- ------------------------ -->
+## Prepare a virtual environment
+Duration: 2
+
+### Prepare a project directory
+
+1. **Set up your project directory**: From the command line in your terminal window, make a directory for your project, for example, called `viam-rover`. And navigate into the directory.
+    ```bash
+    mkdir viam-rover
+    cd viam-rover
+    ```
+
+### Prepare a virtual environment
+[Set up a Python virtual environment](https://docs.viam.com/sdks/python/python-venv/) to avoid any conflicts with other projects or your system.
+1. From the command line in your terminal window, install [`virtualenv` using `pip`](https://virtualenv.pypa.io/en/latest/installation.html#via-pip)
+    ```bash
+    python3 -m pip install --user virtualenv
+    ```
+1. **Create and activate a virtual environment**: In the project directory `viam-rover`, create and activate a virtual environment called `.venv` for Python to run in.
+    ```bash
+    python3 -m venv .venv
+    source .venv/bin/activate
+    ```
+Now, `.venv` prepends the commands in your terminal window to indicate the Python packages being used are from this particular environment. You can exit this environment by running `deactivate`.
+
+### Install the Python SDK
+
+Install the Viam Python SDK and all required general dependencies.
+
+1. **Install the SDK**: From the command line in your terminal window (inside the activated `.venv` Python environment), install the Viam Python SDK:
+    ```bash
+    pip3 install viam-sdk
+    ```
+
+    > aside negative
+    > Windows is not supported. If you are using Windows, use the [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install) and install the Python SDK using the preceding instructions for Linux. For other unsupported systems, see [Installing from source](https://python.viam.dev/#installing-from-source).
+
+<!-- ------------------------ -->
+## Using variables in local development
+Duration: 5
+
+### Copy the sample code
+1. **Refer to Python sample code**: In the Viam app, go to the **CONNECT** tab, and select **Python**. 
+  ![Connect tab](assets/Connect.png)
+1. **Show secrets**: The sample code shows you how to authenticate and connect to a machine, as well as some of the methods you can use on your configured components and services. To show your machine’s API key and API key ID in the sample code, toggle **Include secret** on the CONNECT tab’s Code sample page. 
+  ![Code sample](assets/Code.png)
+1. **Hide secrets**: Instead of leaving our sensitive credentials hard coded in this code sample, let's set up our API key and API key ID as environment variables. Toggle **Include secrets** once again, this time to remove the credentials from the code sample. 
+1. **Copy the code**: From the command line in your terminal window, create a new file called `square.py`, and copy the code sample into this file.
```

Comment:
This almost makes it sound like `touch square.py` is going to copy the contents of the code sample into the file. 

Maybe one step should be making the `square.py` file (why is it called that btw?), followed by a step to instruct copying the contents of the code sample to that new file.

Reply:
Separated into 2 steps, adding in a different commit.

I was following a different tutorial, but then simplified to remove the "drive rover in a square" code. I updated to `rover.py` file name, in a different commit.